src/lib/api.js

```diff
@@ -96,6 +96,66 @@ const api = {
     return request({ uri: `${environment.instanceUrl}/api/deploy-info`, method: 'GET', json: true }) // endpoint added in 3.5
       .then(deploy_info => deploy_info && deploy_info.version);
   },
+
+  /**
+   * Whether form validation endpoint exists or not, by
+   * default we assume it exists, but once `formsValidate`
+   * is called if the response is a 404 error, the
+   * value is changed to `false`, so next call to
+   * the function the request is omitted and the
+   * form considered valid
+   */
+  _formsValidateEndpointFound: true,
+
+  /**
+   * Validates an XForm against the API.
+   * @param formXml XML string
+   * @returns a JSON object if the validation is successful,
+   *          typically `{ok: true}`. If the validation endpoint
+   *          does not exist, the form is considered valid
+   *          and `{ok: true, formsValidateEndpointFound: false}`
+   *          is returned.
+   *          If the method is called again after the endpoint
+   *          was not found, `{ok: true, formsValidateEndpointFound: false}`
+   *          will be returned again without calling the API
+   * @throws `Error` exception with the validations error message
+   *         from the API
+   */
+  formsValidate(formXml) {
+    if (!this._formsValidateEndpointFound) {
+      // The endpoint to validate forms doesn't exist in the API,
+      // (old version), so we assume form is valid but return special result
+      return Promise.resolve({ok: true, formsValidateEndpointFound: false});
+    }
+    return request({
+      method: 'POST',
+      uri: `${environment.instanceUrl}/api/v1/forms/validate`,
+      headers: { 'Content-Type': 'application/xml' },
+      body: formXml,
+    })
+    .then(resp => {
+      try {
+        return JSON.parse(resp);
+      } catch (e) {
+        new Error('Invalid JSON response validating XForm against the API: ' + resp);
```

Comment:
:facepalm: yes this is the block I added to check the JSON response later in the PR and because I couldn't test it easily (though I could just try to change the response in my local CHT env, not so hard), I should at least add a test case. I'll add it now.